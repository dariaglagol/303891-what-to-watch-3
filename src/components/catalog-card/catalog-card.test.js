import React from "react";
import renderer from "react-test-renderer";
import CatalogCard from "./calalog-card";

const MOCKED_FILM = {
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  posterUrl: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  genre: `Comedy`,
  preview: `preview`,
};

const MOCKED_PLAY_PROP = false;

const MOCKED_MUTE_SOUND = true;

it(`Render catalog card`, () => {
  const catalogCardComponent = renderer
    .create(
        <CatalogCard
          film={MOCKED_FILM}
          onFilmCatalogCardHover={() => {}}
          onFilmClick={() => {}}
          muteSound={MOCKED_MUTE_SOUND}
          isPlaying={MOCKED_PLAY_PROP}
        />, {createNodeMock: () => {
          return {};
        }}
    )
    .toJSON();

  expect(catalogCardComponent).toMatchSnapshot();
});
