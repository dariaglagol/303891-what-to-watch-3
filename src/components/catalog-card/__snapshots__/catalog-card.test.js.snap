// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render catalog card 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    href="/films/1"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
    />
    <h3
      className="small-movie-card__title"
    >
      <span
        className="small-movie-card__link"
      >
        name
      </span>
    </h3>
  </a>
</article>
`;
